File: 01_Greedy_기출문제/03_문자열뒤집기.py
arr = input()

cnt0 = 0
cnt1 = 0
cur_char = arr[0]

if cur_char == '0' :
    cnt0 += 1
else :
    cnt1 += 1

for i in range(0, len(arr)) :
    if cur_char != arr[i] :
        if arr[i] == '0' :
            cnt0 += 1
        else :
            cnt1 += 1
        cur_char = arr[i]

print(min(cnt0, cnt1))

'''

0001100

0101010001110

'''

File: 01_Greedy_기출문제/03_문자열뒤집기.cpp
// 문자열을 완전탐색하여
// 0이 연속적으로 있는 부분이 나타나는 횟수를 cnt1
// 1이 연속적으로 있는 부분이 나타나는 횟수를 cnt2
// min(cnt1, cnt2) 출력

#include <iostream>
#include <string>

using namespace std;

int main(void){

    string s;
    getline(cin, s);

    int cnt[2] = {0};
    char cur_char = s[0];
    cnt[cur_char - '0'] = 1;

    for(int i = 0 ; i < s.length() ; i++){
        if(cur_char != s[i]){
            cnt[s[i] - '0'] += 1;
            cur_char = s[i];
        }
    }
    cout << min(cnt[0], cnt[1]);
    
    return 0;
}

/*

0001100  --> 1

0101010001110 --> 4

*/